src/amber/cli/templates/app/public/dist/stylesheets-4bc3cc12d2780fec8bb311892094977b.css

```diff
@@ -66,7 +66,7 @@ h6, .h6 {
 
 #logo {
   background-color: #f4994b;
-  background-image: url(/dist/images/logo.png);
+  background-image: url(/dist/images/logo-c321a05deb5e6e097a40ecffbf90de45.png);
```

Comment:
There are a bunch of rendered files in here, compressed/mapped, etc. I would think we don't need to track those in git since the original sources will be there. The compiled assets will just get compiled on the first run of `amber w`, right?

Reply:
@robacarp  We were doing this because we didn't have `npm` integrated with `amber w` but I think we still want this.  When you first startup, it takes a while for npm to compile and run.  wdyt?

Reply:
Since this pull auto retries, I don’t think it creates a race condition anymore.

Reply:
ok, so for ~5 seconds the styles will not show but once the `npm` finishes, the auto reload will kick in and fix it.  Is that correct?

What if they don't have `npm` installed?

Reply:
@drujensen the way I read this code, it won't even compile until there is a manifest.json file available, so the amber server won't even be available. There will simply be a delay until npm renders the manifest. 

What about on a slower computer (think rasbpi), or a project with a huuuuuuuuge asset bundle, maybe npm doesn't finish building manifest.json until after the 5s timeout is hit. @JoshJuncker Any idea if thats a risk?

As for a dependency on npm, Amber ships by default using webpack to build assets. If you don't have npm running, I'm not sure the default project structure even makes sense. `src/assets` files don't ever get copied out to `/public` and so the static handler never serves them. Shipping precompiled static assets is a nice wedge to make the developer onboarding easier, especially before `amber w` was rewritten to spawn npm (what happens during `amber w` if npm insn't installed?)

Reply:
> what happens during amber w if npm insn't installed?

@robacarp Currently Amber just outputs `--: npm: command not found`, and works without problem: (without webpack)

![ezgif-4-8e25e1f244](https://user-images.githubusercontent.com/3067335/35365380-477bd558-0142-11e8-80e3-c2389c39614a.gif)

Reply:
@faustinoaq thanks for the nice demo.

I guess I'm a little divided. I don't really want Amber to depend on npm, but right now it does and I think acknowledging that would be more helpful to newcomers than trying to disguise it.

Reply:
> I don't really want Amber to depend on npm

@robacarp That's why #476 is important. Making `amber watch` configurable will allow us to use npm alternatives.

Reply:
If you want compiling/precompiling assets built in, is there a way around not depending on something to do that compiling?  Whether it be npm with webpack or if someone makes all the tools needed in crystal, it will still be a dependency.  The way it is structured right now, you can leave npm out.  I removed the necessity of having a manifest.json.  It will only require it if you use the asset helpers.  But since the default app template does use asset_path for including the base javascript and css, I have included the compiled assets and manifest so the project won't fail to run on the first amber watch.  If they want to remove it, they definitely  can.  I guess I would call it a soft dependency? :).  
The hard dependency is as @faustinoaq mentioned in amber watch.  As far as I know, npm and webpack are being run in amber watch no matter what.  I could be wrong though as I'm new... :).  
My opinion on the matter is to have a good default for compiling assets similar to sprockets.  For now, I think npm with webpack is a good option as it seems to be the best current standard.

Reply:
@JoshJuncker I agree with you and I think we should leave in the compiled assets in the repo so the startup time is quick.  Also, if they don't have npm installed, it will still work, albeit they won't be able to make changes.  :-)

We will remove the dependency on npm in amber watch.  It should be configurable so they can replace it with yarn or any other favorite js toolset.

npm is a good starting point and should handle everything that sprockets provided.  This should make it just a little bit easier for a Rails project to be ported over.

Reply:
> @robacarp That's why #476 is important. Making amber watch configurable will allow us to use npm alternatives.

Whatever we use I think that it should be default and supported. Not just a random example of "this is how you could do it.."

> @robacarp I don't really want Amber to depend on npm

It would be very nice if we could do all of this in crystal rather than relying on any sort of node dependancy. Obviously we could, I just don't think any of currently have time to replicate sass, scss, es6 compilation and auto reloading... although honestly the last one is probably the easiest with crystal.

Reply:
Yeah, ES6 compilation - or trans-piling as I've heard it called. I don't think anyone wants to implement that.